osf/models/node.py

```diff
@@ -139,7 +140,7 @@ def can_view(self, user=None, private_link=None):
 
             qs |= self.filter(private_links__is_deleted=False, private_links__key=private_link)
 
-        if user is not None:
+        if user is not None and not isinstance(user, AnonymousUser):
```

Comment:
We should probably be using `user.is_anonymous` here, but we do this type-checking in other places, so it doesn't necessarily have to happen now.